.github/workflows/benchmark.yml

```diff
@@ -27,6 +27,14 @@ jobs:
           node-version: "18.18.0"
           cache: 'yarn'
 
+      - name: Cache nx
+        uses: actions/cache@v3.3.2
+        with:
+          path: ./.nx-cache
+          key: ${{ runner.os }}-nx
+          restore-key: |
+            ${{ runner.os }}-nx
+
```

Comment:
This will never update even if the code changes under the hood...

Reply:
when the code changes, and the cache is no longer a hit, nx recomputes it.

Reply:
Right, but GitHub refuses to upload a new artifact if there is an existing one that matches the same key.

Reply:
@nolanlawson 
I did the following tests locally:

Test 1: added changes to src

1. yarn build
2. yarn build  => verified it reused the cached files
3. added a change to `./packages/src/@lwc/engine-server/polyfills.ts` (just added a new line in one of the comments and saved it)
4. yarn build => cache hit was not found for engine-server and nx rebuilt it


Test 2: added changes to `./.npx-cache`

1. yarn build
2. yarn build  => verified it reused the cached files
3. deleted a cache folder and its correspondent `*.commit` file for the `engine-server` package
4. yarn build => cache hit was not found for engine-server and nx rebuilt it


with these 2 tests I concluded I could just keep updating the same cache on every build and just add the new folders to the cache when there is a non cache hit

Reply:
Right, but you did this all on the same machine, not in GitHub Actions, correct?

Reply:
Using the same key I thought the `.nx-cache` would accumulate new folders on new builds like so:

````bash
.nx-cache
├── 1
├── 2
├── 1.commit
└── 2.commit

.nx-cache
├── 1
├── 2
├── 3
├── 1.commit
├── 2.commit
└── 3.commit

.nx-cache
├── 1
├── 2
├── 3
├── 1.commit
├── 2.commit
└── 3.commit

.nx-cache
├── 1
├── 2
├── 3
├── 4
├── 1.commit
├── 2.commit
├── 3.commit
└── 4.commit
````

Reply:
No, if there's a cache hit, then GitHub Actions does not re-upload to overwrite the cache, even if the underlying directory/file structure changed.

Reply:
It wouldnt work. I just found out that a cache is immutable. :/